common/src/main/java/org/opensearch/ml/common/connector/OciConnector.java

```diff
@@ -0,0 +1,132 @@
+
+    @Builder(builderMethodName = "ociConnectorBuilder")
+    public OciConnector(String name, String description, String version, String protocol,
+                        Map<String, String> parameters, Map<String, String> credential, List<ConnectorAction> actions,
+                        List<String> backendRoles, AccessMode accessMode, User owner) {
+        super(name, description, version, protocol, parameters, credential, actions, backendRoles, accessMode, owner);
+        validate();
+    }
+
+    public OciConnector(String protocol, XContentParser parser) throws IOException {
+        super(protocol, parser);
+        validate();
+    }
+
+
+    public OciConnector(StreamInput input) throws IOException {
+        super(input);
+        validate();
+    }
+
+    public OciConnector(String protocol, StreamInput input) throws IOException {
+        super(protocol, input);
+        validate();
+    }
+
+    private void validate() {
+        if (parameters == null) {
+            throw new IllegalArgumentException("Missing credential");
+        }
+        if (!parameters.containsKey(AUTH_TYPE_FIELD)) {
+            throw new IllegalArgumentException("Missing auth type");
+        }
+
+        final OciClientAuthType authType =
+                OciClientAuthType.from(
+                        parameters.get(AUTH_TYPE_FIELD).toUpperCase(Locale.ROOT));
+
+        if (authType == OciClientAuthType.USER_PRINCIPAL) {
+            if (!parameters.containsKey(TENANT_ID_FIELD)) {
+                throw new IllegalArgumentException("Missing tenant id");
+            }
+
+            if (!parameters.containsKey(USER_ID_FIELD)) {
+                throw new IllegalArgumentException("Missing user id");
+            }
+
+            if (!parameters.containsKey(FINGERPRINT_FIELD)) {
+                throw new IllegalArgumentException("Missing fingerprint");
+            }
+
+            if (!parameters.containsKey(PEMFILE_PATH_FIELD)) {
+                throw new IllegalArgumentException("Missing pemfile");
+            }
+
+            if (!parameters.containsKey(REGION_FIELD)) {
+                throw new IllegalArgumentException("Missing region");
+            }
+        }
+    }
+
+    @Override
+    public Connector cloneConnector() {
+        try (BytesStreamOutput bytesStreamOutput = new BytesStreamOutput()){
+            this.writeTo(bytesStreamOutput);
+            final StreamInput streamInput = bytesStreamOutput.bytes().streamInput();
+            return new OciConnector(streamInput);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * The type of authentication supported by OCI. For more details please visit doc
+     * https://docs.public.oneportal.content.oci.oraclecloud.com/en-us/iaas/Content/API/Concepts/sdk_authentication_methods.htm
+     */
+    public enum OciClientAuthType {
+        RESOURCE_PRINCIPAL,
+        INSTANCE_PRINCIPAL,
+        USER_PRINCIPAL;
```

Comment:
From the doc does USER_PRINCIPAL map to API Key based auth? 
I would recommend using the same name from OCI documentation. 
Also why is Sessions token based auth not listed here?

Reply:
Yeah, in OCI, we often call it USER_PRINCIPAL (and it is more familiar for us), I added a comment to make it clearer

```
Also known as API Key-based authentication
```

Also added a comment why we don't support session token

Session Token-Based Authentication is not supported since it has a short lifetime, and it requires manual refresh through user manual login